nimbus-consensus/src/lib.rs

```diff
@@ -247,19 +248,29 @@ where
 
 pub(crate) fn seal_header<B>(
 	header: &B::Header,
-	keystore: &dyn SyncCryptoStore,
-	type_public_pair: &CryptoTypePublicPair,
+	keystore: &dyn Keystore,
+	public_pair: &Vec<u8>,
+	crypto_algorithm: &u8
```

Comment:
Cant we pass CryptoTypeId as the type parameter here?

Reply:
Yes we can and it is a nice approach as well. I though of passing a numeric option as the function currently only admits that 3 kind of algorithms, but I will change it for it to be more parameterized